bigtable/client/metadata_update_policy_test.cc

```diff
@@ -13,13 +13,278 @@
+      google::bigtable::admin::v2::Table* response) override {
+    for (std::multimap<grpc::string_ref, grpc::string_ref>::const_iterator it =
+             context->client_metadata().begin();
+         it != context->client_metadata().end(); ++it) {
+      auto ele = std::make_pair((*it).first, (*it).second);
+      client_metadata_.emplace(ele);
+    }
+    return grpc::Status::OK;
+  }
+
+  grpc::Status DeleteTable(
+      grpc::ServerContext* context,
+      google::bigtable::admin::v2::DeleteTableRequest const* request,
+      ::google::protobuf::Empty* response) override {
+    for (std::multimap<grpc::string_ref, grpc::string_ref>::const_iterator it =
+             context->client_metadata().begin();
+         it != context->client_metadata().end(); ++it) {
+      auto ele = std::make_pair((*it).first, (*it).second);
+      client_metadata_.emplace(ele);
+    }
+    return grpc::Status::OK;
+  }
+
+  grpc::Status GetTable(
+      grpc::ServerContext* context,
+      google::bigtable::admin::v2::GetTableRequest const* request,
+      google::bigtable::admin::v2::Table* response) override {
+    for (std::multimap<grpc::string_ref, grpc::string_ref>::const_iterator it =
+             context->client_metadata().begin();
+         it != context->client_metadata().end(); ++it) {
+      auto ele = std::make_pair((*it).first, (*it).second);
+      client_metadata_.emplace(ele);
+    }
+    return grpc::Status::OK;
+  }
+
+  const std::multimap<grpc::string_ref, grpc::string_ref>& client_metadata()
+      const {
+    return client_metadata_;
+  }
+
+ private:
+  std::multimap<grpc::string_ref, grpc::string_ref> client_metadata_;
+};
+
+/// The implementation of EmbeddedServer.
+class DefaultEmbeddedServer {
+ public:
+  explicit DefaultEmbeddedServer() {
+    int port;
+    std::string server_address("[::]:0");
+    builder_.AddListeningPort(server_address, grpc::InsecureServerCredentials(),
+                              &port);
+    builder_.RegisterService(&bigtable_service_);
+    builder_.RegisterService(&admin_service_);
+    server_ = builder_.BuildAndStart();
+    address_ = "localhost:" + std::to_string(port);
+  }
+
+  std::string address() const { return address_; }
+  void Shutdown() { server_->Shutdown(); }
+  void Wait() { server_->Wait(); }
+  const std::multimap<grpc::string_ref, grpc::string_ref>&
+  admin_service_client_metadata() const {
+    return admin_service_.client_metadata();
+  }
+  const std::multimap<grpc::string_ref, grpc::string_ref>&
+  bigtable_service_client_metadata() const {
+    return bigtable_service_.client_metadata();
+  }
+
+ private:
+  BigtableImpl bigtable_service_;
+  TableAdminImpl admin_service_;
+  grpc::ServerBuilder builder_;
+  std::unique_ptr<grpc::Server> server_;
+  std::string address_;
+};
+
+/// @test A test for setting metadata when table is not known.
+TEST(MetadataUpdatePolicy, RunWithEmbeddedServer) {
+  std::unique_ptr<DefaultEmbeddedServer> server =
```

Comment:
There is a lot of common initialization in these tests can that be moved to a fixture?

Reply:
created embedded_server_test_fixture for this.